### PR TITLE
headline generation example 

### PR DESCRIPTION
**Pull request by sallamander**
I'm currently finishing up a project generating headlines from news article body text. Given the recent interest in sequence to sequence generation, I wondered if a smaller, lightweight version of this project would be a welcome PR for the `examples`. 

In order to keep it lightweight, quickly trainable, and not download a new text dataset, I was just going to create 4-5 body/headline pairs myself (2 of which are in this PR as an example), and then work through the entire sequence to sequence process. I believe that all of the textual datasets in Keras already come encoded, and I thought showing the complete process end to end might be nice. 

Would this be a PR that would be welcome?

### CODE REVIEW

**Comment by fchollet** [comment]
Yes, that sounds interesting. You would need to make the script very simple and very short, with a lot of clear comments, and you would need to package your data into something that can be readily downloaded.

The data would also need to be in the public domain.

**Comment by sallamander** [comment]
This is basically what I had in mind, although I'm sure some changes will need to be made. 

My idea was just to use a toy dataset to keep it extremely light. This script runs in just about 30 seconds, and was built to overfit to the training data just to be an example. While we could put an external data set that is publicly available in (20 news groups would be possible), I think keeping it simple and fast like this would be nice. 

I did my best to be verbose and clear with the comments, and ran them past some colleagues to try to touch them up and improve them. I'm happy to iterate on the comments as well as the code.

**Comment by fchollet** [comment]
> My idea was just to use a toy dataset to keep it extremely light. This script runs in just about 30 seconds, and was built to overfit to the training data just to be an example. 

Learning to copy and paste input data isn't particularly interesting; this should be trained on a real dataset and properly evaluated on a test set.

**Comment by sallamander** [comment]
I'm going to go ahead and close this PR. Training on a real dataset in a reasonable amount of time means taking a subset of the 20 news dataset, which leads to a small enough data set that's it's pretty difficult to get reasonable predictions on the test set (the code that currently exists on this branch I think  demonstrates this). I'm closing this PR until I can either get some transfer learning set up, or otherwise figure out how to get reasonable predictions on the test set with this small of a data set. I might look into using a different data set as well, but that might take some time.

**Comment by krishnateja614** [comment]
@sallamander  I'm currently working on a similar problem and it would be helpful for me if your code is available publicly.

**Comment by sallamander** [comment]
@krishnateja614 I got a notification about a comment you left somewhere (thought it was here, but can't seem to locate it anywhere). In any case, All the code for my project can be found here... 

https://github.com/sallamander/headline-generation

Feel free to send any comments or anything via that repo.
